### ISSUE TITLE
Hidden input element should be ignored by screen readers

### ISSUE THREAD

**Issue by driskull** [Esri/calcite-design-system#11419]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

Hidden form elements are read by screen readers

![Image](https://github.com/user-attachments/assets/d3166db7-bbb4-4531-80e7-eaec671615bd)

### Expected Behavior

Hidden input element should be ignored by screen readers

### Reproduction Sample

https://codepen.io/Matt-Driscoll/pen/MYgxmLV?editors=1010

### Reproduction Steps

1. open the codepen
2. run axe validation on the page
3. notice the error reported regarding the hidden input
4. The hidden form input does not have an aria-hidden attribute

<img width="786" alt="Image" src="https://github.com/user-attachments/assets/2409ce6e-6255-424b-a6bf-e0adf919b8c1" />

### Reproduction Version

3.0.0-next

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Maps SDK for JavaScript
